### ISSUE TITLE
Allow Sound to still play on Vibrate mode in Android but not in Silence Mode

### ISSUE THREAD

**Issue by nino-finch** [bluefireteam/audioplayers#950]
Right now when we set `respectSilence=true` in the code - it'll make the library stop playing sound when the android phone is set to either silence or vibrates only.  We've gotten few feedbacks from our users that the mental model for android users is to not have "Vibrate" mode considered as "Silence" mode so would be great if this library can support that too.

**Comment by Gustl22** [comment]
Have a look at [`AudioContextAndroid`](https://pub.dev/documentation/audioplayers_platform_interface/latest/audioplayers_platform_interface/AudioContextAndroid-class.html), where you can set the `usageMode` more granularily.

```
AudioContextAndroid(
      usageType: respectSilence
          ? AndroidUsageType.notificationRingtone
          : AndroidUsageType.media,
    );
```

For more see https://developer.android.com/reference/android/media/AudioAttributes.Builder#setUsage(int)

**Comment by corey-sobel** [comment]
@Gustl22 to clarify is there a usage type that supports playing audio on Vibrate Mode & Standard / Ringer Mode but not on silent mode? 

I'm using `audioplayers` version 5.2.0. I verified the behavior is still as above where if `respectSilence` is set to `true` then `AndroidUsageType.notificationRingtone` is used and if it is set to `false` then `media` is used. 

However `notificationRingtone` only plays audio if the phones audio mode is set to ring where as `media` plays audio even if the phones audio mode is set to silent. This is true if I use `PlayerMode.mediaPlayer` and `PlayerMode.lowLatency`.

I tried using `AndroidUsageType.game` as in the below code snippet
`audio_players.AudioContext(
      iOS: config.buildIOS(),
      android: config
          .buildAndroid()
          .copy(usageType: audio_players.AndroidUsageType.game),
    );`

But I observed that has the same behavior as `AndroidUsageType.media`. Is there an example of an `AudioContext` that will play audio in both "Ring" and "Vibrate" audio modes for the phone?
